src/core/string_set.cc

```diff
@@ -129,6 +115,29 @@ uint32_t StringSet::ObjExpireTime(const void* str) const {
   return absl::little_endian::Load32(ttlptr);
 }
 
+void StringSet::ObjUpdateExpireTime(const void* obj, uint32_t ttl_sec) {
+  DCHECK_GT(ttl_sec, 0u);  // ttl_sec == 0 would mean find and delete immediately
```

Comment:
This is used by both `StringSet` && `StringMap`. Maybe worth extracting this into a common function and call it here to avoid code duplication ?

Reply:
Very doable, but where would that common code live? I'm not so familiar with the code-base :)

Reply:
Well this just updates the Expiry time of an `sds` object. A free standing function / helper that accepts a `stride` parameter (the ` + 8` in the example). Or even a function within `DenseSet`. (Although personally I prefer helpers to be defined outside of a class but again that's a preference)

Reply:
I put this in `sds_util::SdsUpdateExpireTime` for now. It might have implications for `ObjUpdateExpireTime` too, but I'll defer that change until we're in agreement on all the other points :)